zettarepl/transport/progress_report_mixin.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding=utf-8 -*-
+import logging
+import re
+import threading
+
+from zettarepl.utils.shlex import implode
+
+from .interface import *
+
+logger = logging.getLogger(__name__)
+
+__all__ = ["ProgressReportMixin"]
+
+
+class ProgressReportMixin:
+    def _get_send_shell(self):
+        raise NotImplementedError
+
+    def _zfs_send_can_report_progress(self):
+        send_shell = self._get_send_shell()
+
+        try:
+            send_shell.exec(["zfs", "send", "-V"])
+        except ExecException as e:
+            if "missing snapshot argument" in e.stdout:
+                # Option is supported (patched zfs on FreeNAS)
+                return True
+            else:
+                # invalid option 'V'
+                return False
+        else:
+            return False
+
+    def _wrap_send(self, send):
+        return ["sh", "-c", "(" + implode(send) + " & PID=$!; echo \"zettarepl: zfs send PID is $PID\" 1>&2; "
+                            "wait $PID)"]
+
+    def _start_progress_observer(self):
+        self.stop_progress_observer = threading.Event()
+
+        pid = self.async_exec.head(self._get_zettarepl_pid, 10)
+
+        threading.Thread(daemon=True, name=f"{threading.current_thread().name}.progress_observer",
```

Comment:
Setting thread name does not work on BSD IIRC, do we want to use try to py-bsd if available?

Reply:
I don't need this in kernel, I use it in python logger